specs/monitoring_integration_service.adoc

```diff
@@ -0,0 +1,173 @@
+
+Namespace:
+
+A manager will create its own namespace for the monitoring-integration to separate
+its own objects from the other modules. These objects are only accessible by
+monitoring-integration.
+
+Object:
+
+Each tendrl component has its own set of objects for a different purpose. Each
+object have its own structure. Monitoring-integration have an object called Config.
+
+Install and run:
+
+Monitoring-integration will be installed in service machine only. It will be
+installed and started by tendrl-ansible as a service. There is just one instance
+of monitoring-integration installed across all clusters.
+
+== Alternatives
+
+None
+
+== Data model impact:
+
+Monitoring-integration configuration stored in etcd : “_NS/monitoring_integration/config”
+
+== Impacted Modules:
+
+=== Tendrl API impact:
+
+None
+
+=== Notifications/Monitoring impact:
+
+None
+
+=== Tendrl/common impact:
+
+None
+
+=== Tendrl/node_agent impact:
+
+None
+
+=== Sds integration impact:
+
+None
+
+
+=== Tendrl/monitoring-integration impact:
+
+Changes need to convert monitoring-integration as service:
+
+* Create a module called manager in monitoring-integration
+* Create a namespace for the monitoring-integration
+* Create module called objects to place all monitoring-integration objects.
+
+The implementation of these three modules is in the implementation section.
+
+=== Security impact:
+
+None
+
+=== Other end user impact:
+
+None
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+Convert the monitoring-integration as service
+
+== Implementation:
+
+Steps to change monitoring-integration as service:
+
+* Create a module called manager in tendrl/monitoring_integration.
+* Create a file called "__init__.py" in tendrl/monitoring_integration/manager.
+* A file "__init__.py" is the entry point for monitoring-integration. Create a
+  main function in this file.
+* The main function which is present at manager module will initiate the flow.
+* Create a class called "MonitoringIntegrationManager" in "__init__.py" which is
+  used to initialize all gevent classes.
+* A class MonitoringIntegrationManager should be inherited from the base class called
+  “Commons_manager.Manager”.
+* Create a class called "MonitoringIntegrationNS" in tendrl/monitoring_integration/__init__.py
+  to create a namespace.
+* A class "MonitoringIntegrationNS" should be a subclass of "TendrlNS".
+* Create a module called objects in tendrl/monitoring_integration/
+* Create an object called "Config" in tendrl/monitoring_integration/objects
+* The main function in manager should create the namespace using "MonitoringIntegrationNS"
+  class before actual flow starts.
+* The main function in manager should initialize and start the all gevent classes using "MonitoringIntegrationManager"
+  class.
+* The manager should stop all gevents when monitoring-integration service is stopped.
+
```

Comment:
You need the tendrl-monitoring-integration.service systemd unit, copy it from the performance-monitoring repo and make changes